pkg/server/node.go

```diff
@@ -2064,6 +2078,64 @@ type streamManager interface {
 	Error() chan error
 }
 
+// perConsumerLimiter is a ConcurrentRequestLimiter for a given mux rangefeed
+// consumer. It is stored in the Node as long as there are active MuxRangeFeed
+// requests the related ConsumerID.
+type perConsumerLimiter struct {
+	l limit.ConcurrentRequestLimiter
+	c int
+}
+
+func (n *Node) perConsumerCatchupScanLimiter(
+	consumerID int64, sv *settings.Values,
+) *limit.ConcurrentRequestLimiter {
+	n.perConsumerCatchupLimiterMu.Lock()
+	defer n.perConsumerCatchupLimiterMu.Unlock()
+	if _, ok := n.perConsumerCatchupLimiterMu.limiters[consumerID]; ok {
+		n.perConsumerCatchupLimiterMu.limiters[consumerID].c++
+	} else {
+		n.perConsumerCatchupLimiterMu.limiters[consumerID] = &perConsumerLimiter{
+			l: makePerConsumerScanLimiter(consumerID, sv),
+			c: 1,
+		}
+	}
+	return &n.perConsumerCatchupLimiterMu.limiters[consumerID].l
+}
+
+func (n *Node) releasePerConsumerCatchup(consumerID int64) {
+	n.perConsumerCatchupLimiterMu.Lock()
+	defer n.perConsumerCatchupLimiterMu.Unlock()
+	if l, ok := n.perConsumerCatchupLimiterMu.limiters[consumerID]; ok {
+		l.c--
+		if l.c == 0 {
+			delete(n.perConsumerCatchupLimiterMu.limiters, consumerID)
+		}
+	}
+}
+
+func makePerConsumerScanLimiter(
+	consumerID int64, sv *settings.Values,
+) limit.ConcurrentRequestLimiter {
+	getLimit := func() int {
+		// Set the limit to 75% of the total limit.
+		return min(int(float64(kvserver.ConcurrentRangefeedItersLimit.Get(sv))*0.75), 1)
+	}
+	l := limit.MakeConcurrentRequestLimiter(
+		fmt.Sprintf("PerConsumerCatchupLimit-%d", consumerID),
+		getLimit())
+	kvserver.ConcurrentRangefeedItersLimit.SetOnChange(sv, func(ctx context.Context) {
+		l.SetLimit(getLimit())
+	})
+	return l
+}
+
+// defaultRangefeedConsumerID returns a random ConsumerID. Used by
```

Comment:
I think this ID will be "unique enough". The ID is created using a combination of the timestamp and the instance ID.  Within a cluster this should be unique. We _may_ allow a consumer to send this all the way from another cluster in which case there is some incredibly small change it would not be unique.